lib/tests/streamlit/elements/vega_charts_test.py

```diff
@@ -121,6 +159,59 @@ def test_empty_altair_chart_throws_error(self):
         with self.assertRaises(TypeError):
             st.altair_chart(use_container_width=True)
 
+    def test_dataset_names_stay_stable(self):
+        """Test that dataset names stay stable across multiple calls
+        with new Pandas objects containing the same data.
+        """
+        # Execution 1:
+        df = pd.DataFrame([["A", "B", "C", "D"], [28, 55, 43, 91]], index=["a", "b"]).T
+        chart = alt.Chart(df).mark_bar().encode(x="a", y="b")
+        st.altair_chart(chart)
+        chart_el_1 = self.get_delta_from_queue().new_element
+
+        # Execution 2 (recreate the same chart with new objects)
+        df = pd.DataFrame([["A", "B", "C", "D"], [28, 55, 43, 91]], index=["a", "b"]).T
+        chart = alt.Chart(df).mark_bar().encode(x="a", y="b")
+        st.altair_chart(chart)
```

Comment:
Does this mean that the two `st.altair_chart` widgets have the same computed id or what else is contributing to the altair_chart id?